### PR TITLE
Update module template to support container registries

### PR DESCRIPTION
**Pull request by edmundmiller**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

Does the prep work for https://github.com/nf-core/modules/pull/2291

### CODE REVIEW

File: nf_core/modules/bump_versions.py [hunk 1]
```diff
@@ -41,9 +41,8 @@ def bump_versions(self, module=None, all_modules=False, show_uptodate=False):
         Bump the container and conda version of single module or all modules
 
         Looks for a bioconda tool version in the `main.nf` file of the module and checks whether
-        are more recent version is available. If yes, then tries to get docker/singularity
-        container links and replace the bioconda version and the container links in the main.nf file
-        of the respective module.
+        are more recent version is available. If yes, then tries to get container links and replace
```
Comment (mahesh-panchal) [inline nf_core/modules/bump_versions.py:44]:
```suggestion
        a more recent version is available. If yes, it then tries to get container links and replace
```

**Review by mahesh-panchal — COMMENTED** [review]
Seems the JSON schema is also missing the new param.

**Comment by drpatelh** [comment]
Think we need to iron out how we get `nf-core download` working with these new changes before this is merged. See Slack discussion [here](https://nfcore.slack.com/archives/CJRH30T6V/p1667380740266769) and feature request https://github.com/nextflow-io/nextflow/issues/3340

File: nf_core/utils.py [hunk 2]
```diff
@@ -681,56 +682,22 @@ def get_biocontainer_tag(package, version):
...
-                current_date = None
-                for k in common_keys:
-                    # Get the most recent common image
-                    date = max(all_docker[k]["date"], all_docker[k]["date"])
-                    if docker_image is None or current_date < date:
-                        docker_image = all_docker[k]["image"]
-                        singularity_image = all_singularity[k]["image"]
-                        current_date = date
-                return docker_image["image_name"], singularity_image["image_name"]
+                return response.json()["tags"][0]["name"]
```
Comment (mirpedrol) [inline nf_core/utils.py:694]: I tried with hla-la and I get an older version than for bioconda. Should we check if versions match and downgrade bioconda if they don't?
```
conda (params.enable_conda ? "bioconda::hla-la=1.0.3" : null)
def container_image = "hla-la:1.0.1--h1b026d1_5"
container [ params.container_registry ?: 'quay.io/biocontainers' , container_image ].join('/')
```

File: tests/modules/lint.py [hunk 3]
```diff
@@ -87,3 +104,17 @@ def test_modules_lint_patched_modules(self):
...
     assert len(module_lint.passed) > 0
     assert len(module_lint.warned) >= 0
+
+
+def test_check_bioconda_package(self):
+    bioconda_package = "bioconda::trim-galore=0.6.7"
+    fix_version = False
+    progress_bar = None
+    container_tag = "0.6.7"
+    nf_core.modules.lint.main_nf._check_bioconda_package(
```
Comment (mirpedrol) [inline tests/modules/lint.py:114]:
```suggestion
    module_lint = nf_core.modules.lint.ModuleLint(self.pipeline_dir)
    nf_core.modules.lint.main_nf._check_bioconda_package(
```

File: tests/modules/lint.py [hunk 4]
```diff
@@ -87,3 +104,17 @@ def test_modules_lint_patched_modules(self):
...
+        self,
```
Comment (mirpedrol) [inline tests/modules/lint.py:115]:
```suggestion
        module_lint,
```

File: nf_core/utils.py (hunk 2 above)
Comment (Emiller88) [reply nf_core/utils.py:694]: Can you manually check if there's a matching hla-la version? I was just lazy and hoped asking it for only one image gave you the latest :grimacing: 

I can check that myself.

Comment (Emiller88) [reply nf_core/utils.py:694]: Okay there is a 1.0.3 version, the problem is it got pushed 4 months ago, but the 1.0.1 version with the `_5` build got pushed 3 months ago.

Comment (mirpedrol) [reply nf_core/utils.py:694]: Oh, I see that's an edge case. Then I guess we can let the user decide, but maybe still print a warning if the versions are different.

**Comment by mashehu** [comment]
I think this is now implemented, please reopen @edmundmiller if that is not the case

**Pull request closed without merging**